pkg/image/registry/image/strategy.go

```diff
@@ -112,6 +112,37 @@ func (imageStrategy) clearSignatureDetails(image *api.Image) {
 	}
 }
 
+// signatureStrategy implements behaviour for signatures subset of image.
+type signatureStrategy struct {
+	imageStrategy
+}
+
+// SignatureStrategy is the default logic that applies when updating
+// image' signatures via the REST API.
+var SignatureStrategy = signatureStrategy{Strategy}
+
+// Canonicalize normalizes the object after validation.
+func (signatureStrategy) Canonicalize(obj runtime.Object) {
+}
+
+// PrepareForUpdate clears fields that are not allowed to be set by end users on update.
+// It allows to update only signatures field of the image.
+func (s signatureStrategy) PrepareForUpdate(obj, old runtime.Object) {
```

Comment:
I'd like to see a validation function that enforces that the only thing this can change is the new signature.